From 2b3527fe7b5f3b821a9966f693825e852ab0eab5 Mon Sep 17 00:00:00 2001
From: Siddharth Vadapalli <s-vadapalli@ti.com>
Date: Fri, 7 Feb 2025 12:30:54 +0530
Subject: [PATCH 352/406] FROMLIST: soc: ti: k3-ringacc: include
 platform_device header

Users of "k3-ringacc" APIs may not necessarily be platform devices. Such
users therefore will not include "platform_device.h", resulting in
compilation warnings for the APIs declared in "k3-ringacc.h" which assume
inclusion of "platform_device.h" by its users.

Fix the compilation warnings by including the "platform_device" header
file, which should ideally have been included since commit under Fixes.

Link: https://patch.msgid.link/20250207065123.879863-1-s-vadapalli@ti.com
Fixes: d782298c6f6b ("soc: ti: k3-ringacc: add AM64 DMA rings support.")
Signed-off-by: Siddharth Vadapalli <s-vadapalli@ti.com>
---
 include/linux/soc/ti/k3-ringacc.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/soc/ti/k3-ringacc.h b/include/linux/soc/ti/k3-ringacc.h
index 39b022b92598..6bbb58068048 100644
--- a/include/linux/soc/ti/k3-ringacc.h
+++ b/include/linux/soc/ti/k3-ringacc.h
@@ -8,6 +8,7 @@
 #ifndef __SOC_TI_K3_RINGACC_API_H_
 #define __SOC_TI_K3_RINGACC_API_H_
 
+#include <linux/platform_device.h>
 #include <linux/types.h>
 
 struct device_node;
-- 
2.39.5 (Apple Git-154)

